### ISSUE TITLE
 sequelize.fn doesn't work

### ISSUE THREAD

**Issue by Negaihoshi** [sequelize/sequelize#4695]
My Code is
```js
let maxDate = await db.Order.find({
        attributes: [db.sequelize.fn('MAX', db.sequelize.col('paymentConfirmDate'))],
});
```
SQL log is 
```sql
SELECT MAX(`paymentConfirmDate`) FROM `Orders` AS `Order` LIMIT 1;
```
This code return empty object.
```js
 {dataValues: {},
  _previousDataValues: {},
  _changed: {},
  '$modelOptions':
   { timestamps: true,
     instanceMethods: {},
     classMethods: { associate: [Function] },
     validate: {},
     freezeTableName: false,
     underscored: false,
     underscoredAll: false,
     paranoid: false,
     whereCollection: null,
     schema: null,
     schemaDelimiter: '',
     defaultScope: null,
     scopes: [],
     hooks: {},
     indexes: [],
     name: { plural: 'Orders', singular: 'Order' },
     omitNull: false,
     sequelize:
      { options: [Object],
        config: [Object],
        dialect: [Object],
        models: [Object],
        modelManager: [Object],
        connectionManager: [Object],
        importCache: [Object],
        test: [Object],
        queryInterface: [Object] },
     uniqueKeys: {},
     hasPrimaryKeys: true },
  '$options':
   { isNewRecord: false,
     '$schema': null,
     '$schemaDelimiter': '',
     raw: true,
     attributes: [ [Object] ] },
  hasPrimaryKeys: true,
  __eagerlyLoadedAssociations: [],
  isNewRecord: false }
```
I use this SQL to query
```sql
SELECT MAX(`paymentConfirmDate`) FROM `Orders` AS `Order` LIMIT 1;
```
result is
<img width="200" alt="2015-10-20 8 24 23" src="https://cloud.githubusercontent.com/assets/1733006/10616423/ac9d16f4-7797-11e5-97d1-5ee8c3576875.png">

Why Sequelize code return empty object?

---------
I using new way to query max paymentConfirmDate.
```js
let maxDate = await db.Order.max('paymentConfirmDate')
```
It SQL log is
```sql
SELECT max(`paymentConfirmDate`) AS `max` FROM `Orders` AS `Order`;
```
Sequelize return
```js
Mon Oct 19 2015 22:03:48 GMT+0800 (CST)
```

--------
Sequelize define `paymentConfirmDate: DataTypes.DATE`
I use Sequelize version 3.4.1/3.12.2 with MySQL.
Why `Order.max('paymentConfirmDate')` is work ?

**Comment by subjectix** [comment]
The problem is that the return value will not have a name, you must follow [this pattern](http://docs.sequelizejs.com/en/latest/docs/querying/#attributes). try:
```js
let maxDate = await db.Order.find({
        attributes: [
            // Needs to be an array, the documentation page 
            // has a resolved un-committed bug see: #4660
            [db.sequelize.fn('MAX', db.sequelize.col('paymentConfirmDate')), 'latestPayment'],
        ],
}).then(result => {
       // Not result.latestPayment, see the doc site
       console.log(result.get('latestPayment'));
});

```

**Comment by janmeier** [comment]
@subjectix is right on the money on this one :)

Closing this issue, since it is neither a bug report nor a feature request.

For general sequelize questions, please use [StackOverflow](https://stackoverflow.com/questions/tagged/sequelize.js) or [Google groups](https://groups.google.com/forum/#!forum/sequelize).
